src/tqec/compile/tree/node.py

```diff
@@ -62,7 +62,7 @@ def __init__(
         self._annotations = dict(annotations) if annotations is not None else {}
 
     @staticmethod
-    def _get_children(layer: LayoutLayer | BaseComposedLayer) -> list[LayerNode]:
+    def _get_children(layer: BaseLayer | LayoutLayer | BaseComposedLayer) -> list[LayerNode]:
```

Comment:
```suggestion
    def _get_children(layer: LayoutLayer | BaseComposedLayer) -> list[LayerNode]:
```
This also needs to be fixed.